stable_baselines3/common/vec_env/base_vec_env.py

```diff
@@ -55,10 +54,17 @@ class VecEnv(ABC):
 
     metadata = {"render.modes": ["human", "rgb_array"]}
 
-    def __init__(self, num_envs: int, observation_space: gym.spaces.Space, action_space: gym.spaces.Space):
+    def __init__(
+        self,
+        num_envs: int,
+        observation_space: gym.spaces.Space,
+        action_space: gym.spaces.Space,
+        render_mode: Optional[str] = None,
+    ):
         self.num_envs = num_envs
         self.observation_space = observation_space
         self.action_space = action_space
+        self.render_mode = render_mode
```

Comment:
as quickly discussed in https://github.com/DLR-RM/stable-baselines3/pull/780#issuecomment-1302090826, we could do `self.render_mode = render_mode or "rbg_array"`, no?
and keep the rest of the code?

Reply:
actually, this should be done in `make_vec_env` and when creating the env from a string

Reply:
> actually, this should be done in `make_vec_env` and when creating the env from a string


Yes we can do that when creating from string, but once a gym class instance is passed we can only 'take note' of the render mode I think?

Reply:
>  we can only 'take note' of the render mode I think?

yes and we do the assert only when render is called